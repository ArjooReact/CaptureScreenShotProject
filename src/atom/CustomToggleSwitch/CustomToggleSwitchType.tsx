export interface CustomToggleSwitchType{
    toggleSwitch:()=>void
    onChange:()=> void
    isEnabled:boolean

}