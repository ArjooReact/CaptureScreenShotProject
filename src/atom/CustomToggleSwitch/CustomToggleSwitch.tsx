import React from 'react'
import {Switch} from 'react-native'
import { CustomToggleSwitchType } from './CustomToggleSwitchType'
import Colors from '../../utility/Colors'
const CustomToggleSwitch:React.FC<CustomToggleSwitchType>=({toggleSwitch,onChange,isEnabled})=>{

    return(<Switch
        trackColor={{false: Colors.TRACK_DISABLE_COLOR, true: Colors.TRACK_ENABLE_COLOR}}
        thumbColor={isEnabled ? Colors.THUMB_ENABLE_COLOR : Colors.THUMB_DISABLE_COLOR}
        ios_backgroundColor={Colors.TOGGLE_BG_COLOR}
        onChange={onChange}
        onValueChange={toggleSwitch}
        value={isEnabled}
     >
    </Switch>)
}
export default CustomToggleSwitch