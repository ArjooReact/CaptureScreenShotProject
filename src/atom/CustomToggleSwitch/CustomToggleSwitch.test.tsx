import renderer from 'react-test-renderer';
import {test, jest, describe} from '@jest/globals';
import {expect, it, beforeEach} from '@jest/globals';
import CustomToggleSwitch from './CustomToggleSwitch';
import { View } from 'react-native';
describe('all mock', () => {
  let mockToggleSwitch = jest.fn();
  let mockOnChange= jest.fn()
  let children=()=>{
    return<View></View>
  }
  test('testcase for matching customToggleSwitch Snapshot==>', () => {
    const AppRef = renderer.create(
      <CustomToggleSwitch toggleSwitch={mockToggleSwitch} onChange={mockOnChange} isEnabled={true} />,
    );
    expect(AppRef.toJSON()).toMatchSnapshot();
  });

  test('testcase for checking presence of dom element for customToggleSwitch==>', () => {
    const AppRef = renderer.create(
        <CustomToggleSwitch toggleSwitch={mockToggleSwitch} onChange={mockOnChange} isEnabled={true} />,
    );
    console.log('arzoo test======', AppRef.toJSON());
    let props: any = AppRef.toJSON();
    console.log('Arzoo data...',props)
    expect(AppRef.toJSON()).toBeTruthy();
  });

  test('testcase for checking child element for customToggleSwitch==>', () => {
    const styles = {};
    const AppRef = renderer.create(
        <CustomToggleSwitch toggleSwitch={mockToggleSwitch} onChange={mockOnChange} isEnabled={true} />,
    );
    let props: any = AppRef.toJSON();
    expect(props?.type).toBe('RCTSwitch');
  });

});
