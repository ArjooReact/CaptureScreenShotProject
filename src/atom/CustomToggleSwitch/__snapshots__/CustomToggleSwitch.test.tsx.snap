// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all mock testcase for matching customImage Snapshot==> 1`] = `
<RCTSwitch
  accessibilityRole="switch"
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onTintColor="#81b0ff"
  style={
    [
      {
        "height": 31,
        "width": 51,
      },
      {
        "backgroundColor": "#3e3e3e",
        "borderRadius": 16,
      },
    ]
  }
  thumbTintColor="#f5dd4b"
  tintColor="#767577"
  value={true}
/>
`;

exports[`all mock testcase for matching customToggleSwitch Snapshot==> 1`] = `
<RCTSwitch
  accessibilityRole="switch"
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onTintColor="#81b0ff"
  style={
    [
      {
        "height": 31,
        "width": 51,
      },
      {
        "backgroundColor": "#3e3e3e",
        "borderRadius": 16,
      },
    ]
  }
  thumbTintColor="#f5dd4b"
  tintColor="#767577"
  value={true}
/>
`;
